feat(devcontainers): add devcontainer support to the project

jmylchreest opened the pull request:
Add initial devcontainer support to the project, and some basic linting

jmylchreest committed:
feat(devcontainers): add devcontainer support to the project

Add initial devcontainer support to the project, and some basic linting

ImDevinC commented on mypy.ini:
```diff
@@ -0,0 +1,3 @@
+[mypy]
+python_version = 3.11
```
StreamController is currently using python 3.12

ImDevinC commented on .devcontainer/devcontainer.json:
```diff
@@ -0,0 +1,31 @@
+      "settings": {
+        "python.defaultInterpreterPath": "/workspaces/StreamController/.venv/bin/python",
+        "editor.formatOnSave": true,
+        "python.formatter.provider": "black",
```
This project doesn't use black as the default formatter. Setting this will cause a lot of whitespace changes the first time the save happens. May want to run this by @Core447 to understand if they want a default formatter

Core447 replied:
Thanks for catching this! I'm using autopep8, so PEP 8

jmylchreest committed:
fix: refactor to a dockerfile to make proper use of dbus/usb passthrough etc.

By moving to a dockerfile this provides us with ability to correctly use
the newer versions of adwaita, passthrough of usb devices and wayland,
and run a session dbus easily solving a number of devcontainer issues I
came across on bluefin.

jmylchreest committed:
fix: removing linter/other opinionated settings

Removing the opinionated settings from the devcontainer/IDE env to avoid
undesirable misalignment with existing standard.
